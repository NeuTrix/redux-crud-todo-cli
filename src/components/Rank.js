import React from 'react';
import PropTypes from 'prop-types';
import { FormControl } from 'react-bootstrap';

// ========= Component

const Rank = (props) => {
	let _rank; // task priority 

	const handleChange = (event) => {
		event.preventDefault();
		props.updateRank(props._id, _rank.value);
	};

	return (

		 <FormControl
			bsSize = 'sm'
			componentClass = "select" 
			defaultValue = {props.currRank}
			onChange = { handleChange } 
			inputRef = { (value) => _rank = value } >

			<option value = 'High'>High</option>
			<option value = 'Med'>Med</option>
			<option value = 'Low'>Low</option>

		</FormControl>
	);
}; 

// ========= Props  

Rank.propTypes = {
	currRank: PropTypes.string.isRequired,
	_id: PropTypes.string.isRequired,
	updateRank: PropTypes.func.isRequired
};

Rank.defaultProps = {
	currRank: 'default',	
	_id: 'default',
	updateRank: f => f
};

export default Rank;