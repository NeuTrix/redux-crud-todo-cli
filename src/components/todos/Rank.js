import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

const propTypes = {
	value: PropTypes.string.isRequired,
};

const Rank = (props) => {
	const { value } = props;

	return (
		<TextField
			{...props} // allow props to pass to wrapped component
			fullWidth
			id="new_item_priority"
			label="set rank"
			margin="dense"
			name="rank"
			select
			value={value}
			variant="outlined"
		>
			<option value="High">High</option>
			<option value="Med">Med</option>
			<option value="Low">Low</option>
		</TextField>
	);
};

Rank.propTypes = propTypes;

export default Rank;
