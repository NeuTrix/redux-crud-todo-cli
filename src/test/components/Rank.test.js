/* eslint-env node, mocha */
import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'enzyme';
import Rank from '../../components/todos/Rank';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(
		<Rank
			value="med"
		/>, div,
	);
});

const wrapper = render(
	<Rank
		value="med"
	/>,
);

it('...renders the Rank ', () => {
	expect(wrapper).toMatchSnapshot();
});
