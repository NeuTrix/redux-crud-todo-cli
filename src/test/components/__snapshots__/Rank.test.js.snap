// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`...renders the Rank  1`] = `
<div
  class="MuiFormControl-root-4 MuiFormControl-marginDense-6 MuiFormControl-fullWidth-7"
>
  <label
    class="MuiFormLabel-root-19 MuiFormLabel-filled-23 MuiInputLabel-root-8 MuiInputLabel-formControl-13 MuiInputLabel-animated-16 MuiInputLabel-shrink-15 MuiInputLabel-marginDense-14 MuiInputLabel-outlined-18 Rank-label-1"
    data-shrink="true"
    for="new_item_priority"
  >
    set rank
  </label>
  <div
    class="MuiInputBase-root-46 MuiOutlinedInput-root-33 MuiInputBase-fullWidth-55 MuiInputBase-formControl-47 MuiInputBase-marginDense-53"
  >
    <fieldset
      aria-hidden="true"
      class="MuiPrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline-40"
      style="padding-left:8px"
    >
      <legend
        class="MuiPrivateNotchedOutline-legend-64"
        style="width:0"
      >
        <span>
          ​
        </span>
      </legend>
    </fieldset>
    <div
      class="MuiSelect-root-26"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        class="MuiSelect-select-27 MuiSelect-selectMenu-30 MuiSelect-outlined-29 MuiInputBase-input-56 MuiOutlinedInput-input-41 MuiInputBase-inputMarginDense-57 MuiOutlinedInput-inputMarginDense-42"
        role="button"
        tabindex="0"
      >
        <span>
          ​
        </span>
      </div>
      <input
        id="new_item_priority"
        name="rank"
        type="hidden"
        value="med"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-65 MuiSelect-icon-32"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
